cider/exec/nextgen/operators/AggNode.cpp

```diff
@@ -0,0 +1,197 @@
+
+namespace cider::exec::nextgen::operators {
+// using namespace jitlib;
+static std::string sql_type_name[SQLTypes::kSQLTYPE_LAST] = {"null",
+                                                             "boll",
+                                                             "char",
+                                                             "varchar",
+                                                             "numeric",
+                                                             "decimal",
+                                                             "int32",
+                                                             "int16",
+                                                             "float",
+                                                             "double",
+                                                             "time",
+                                                             "timestamp",
+                                                             "int64",
+                                                             "text",
+                                                             "date",
+                                                             "array",
+                                                             "interval_day_time",
+                                                             "interval_year_month",
+                                                             "int8",
+                                                             "eval_context_type",
+                                                             "void",
+                                                             "cursor",
+                                                             "column",
+                                                             "column_list",
+                                                             "struct"};
+
+AggExprsInfo::AggExprsInfo(SQLTypes st, SQLAgg at, int32_t so, int32_t bs) {
+  sql_types = st;
+  switch (st) {
+    case kSMALLINT:
+    case kINT:
+    case kBIGINT:
+      jit_value_type = jitlib::JITTypeTag::INT64;
+      break;
+    case kFLOAT:
+    case kDOUBLE:
+    case kBOOLEAN:
+    case kCHAR:
+    case kVARCHAR:
+    case kDECIMAL:
+    default:
+      jit_value_type = jitlib::JITTypeTag::INVALID;
+      break;
+  }
+  agg_type = at;
+  start_offset = so;
+  byte_size = bs;
+  // get agg function name
+  agg_name = "nextgen_cider_agg";
+  switch (agg_type) {
+    case SQLAgg::kSUM: {
+      agg_name = agg_name + "_sum_" + sql_type_name[sql_types];
+      break;
+    }
+    default:
+      LOG(FATAL) << "unsupport type";
+      break;
+  }
+}
+
+TranslatorPtr AggNode::toTranslator(const TranslatorPtr& succ) {
+  return createOpTranslator<AggTranslator>(shared_from_this(), succ);
+}
+
+void AggTranslator::consume(context::CodegenContext& context) {
+  codegen(context);
+}
+
+void initialExpersInfo(AggExprsInfoVector& infos, ExprPtrVector& exprs) {
```

Comment:
please move the output argument after the input argument. i.e `initialExpersInfo(ExprPtrVector& exprs, AggExprsInfoVector& infos)`

Reply:
Why not make ```infos``` as a return value? Meanwhile I think named as ```initExprsInfo``` or ```initializeExprsInfo``` is better.